atomix/utils/src/main/java/io/atomix/utils/concurrent/SingleThreadContext.java

```diff
@@ -155,6 +155,13 @@ public Scheduled schedule(
   @Override
   public void close() {
     executor.shutdownNow();
+
+    try {
+      //noinspection ResultOfMethodCallIgnored
+      executor.awaitTermination(30, TimeUnit.SECONDS);
```

Comment:
While it may introduce some delay on shut down, imo I like the idea that we wait until the thread is terminated. 

We may want to add some logging here, as 30 seconds elapsed without termination means we have a single task taking 30 seconds (:exploding_head:), which should not happen.